src/coreclr/jit/gentree.cpp

```diff
@@ -17383,17 +17406,18 @@ bool GenTreeVecCon::IsHWIntrinsicCreateConstant(GenTreeHWIntrinsic* node, simd32
         case NI_Vector256_Create:
         case NI_Vector256_CreateScalar:
         case NI_Vector256_CreateScalarUnsafe:
+        case NI_Vector512_Create:
 #elif defined(TARGET_ARM64)
         case NI_Vector64_Create:
         case NI_Vector64_CreateScalar:
         case NI_Vector64_CreateScalarUnsafe:
 #endif
         {
             // Zero out the simd32Val
-            simd32Val = {};
+            simd64Val = {};
```

Comment:
I wonder if this is "unnecessary expense".

We could reasonably expect `simd32Val` to be handled "ideally" since it was at worst 2x 128-bits and most modern CPUs support dual dispatch of such instructions.

However, with `simd64Val` we're looking at worst 4x 128-bit operations and it won't be typical for them to get pipelined, so it will start adding up some amount of overhead.

Reply:
It's possible of course that the usages of `simd64Val` aren't enough to have measurable impact.

But, it is something that might be worthing seeing the impact around.

Reply:
1. Re "unnecessary expense", do you mean the zeroing out in particular or the fact that we are using `simd64Val`  now
2. Do you have any suggestions on measuring impact?

Reply:
I presume the issue is that we now pass and use simd64_t many places where simd32_t used to be used, and where the high 32 bytes of the new simd64_t are not needed. So there will be additional zeroing/setting as well as call argument passing costs (if passed by value), etc.

Re: 2: the "superpmi-diffs" job will measure throughput impact by counting machine instructions run, which is at least a proxy for runtime performance.

Reply:
This is causing more issues that I expected. On release builds for x86(32 bit), the new `GenTreeVecCon` size exceeds `TREE_NODE_SZ_LARGE`(It's by some small amount-4 bytes but it still does). This triggers following assert https://github.com/dotnet/runtime/blob/main/src/coreclr/jit/gentree.cpp#L309 . This is tricky because assert condition is triggered for Release but not for debug or Checked

Note - for x64, I already had to make a change : https://github.com/dotnet/runtime/pull/80960/files#diff-fba867eda1c745875748370c59da2195c3823b85a17d199444068d93f91c749bL306-L310

Reply:
We could refactor this to be like "integer constants" and have `GenTreeVecConCommon`. We'd then have a "small" and "large" version.

Alternatively, we could handle this like `GenTreeMultiOp` and have `simd64_t` use `compiler->getAllocator(...).allocate`, or even a combination of the two.

I think we have a few options for handling this without changing the largest node size (although doing that would also be possible).

I think it needs weigh in from @BruceForstall on what the "right" direction to pursue would be, however.

------------------

If simd64 constants are the only real blocking thing here, I think we can special case the two we actually need (`Zero` and `AllBitsSet` iirc) and leave the support for actual 64-bit constants for a follow up PR.

Reply:
Ah...I 'hid' the simd64 constants under `#if defined(TARGET_AMD64)`  in the latest version - https://github.com/dotnet/runtime/pull/80960/files#diff-4b681875302594fccc0147e2528a9af1fa9be6f5c86abf522c9700d1ccdb7cf0L3356-L3357. Not sure if this is a valid long term solution but wanted to explore it

I'm open to your other suggestion too and special casing `Zero` and `AllBitsSet` so that we can revisit this later. I'm unsure how this would work. Won't we need  `gtSimd64Val` for lowering? get_zero() would need `Zero`

Reply:
Given they are the only two simd64 constants that would exist, I imagine we'd just track them as `gtSimd32 Zero` and `gtSimd32 AllBitsSet` as a hacky workaround.

This of course wouldn't be applicable to full simd64 constants, but it lets the other support get in and will allow GT_CNS_VEC to be looked at separately.